src/OrchardCore.Modules/OrchardCore.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -151,16 +151,16 @@ private string GenerateUniquePath(string path, AutoroutePart context)
             while (true)
             {
                 var versionedPath = $"{unversionedPath}-{version++}";
-                if (IsPathUnique(versionedPath, context))
+                if (await IsPathUniqueAsync(versionedPath, context))
                 {
                     return versionedPath;
                 }
             }
         }
 
-        private bool IsPathUnique(string path, AutoroutePart context)
+        private async Task<bool> IsPathUniqueAsync(string path, AutoroutePart context)
         {
-            return _session.QueryIndex<AutoroutePartIndex>(o => o.ContentItemId != context.ContentItem.ContentItemId && o.Path == path).CountAsync().GetAwaiter().GetResult() == 0;
+            return (await _session.QueryIndex<AutoroutePartIndex>(o => o.ContentItemId != context.ContentItem.ContentItemId && o.Path == path).CountAsync()) == 0;
```

Comment:
Remove `await` here and return `_session...` directly. The keywords are not necessary when the only call is the return statement.

Reply:
Ah, but look at the end of the statement: it performs a comparison with a 0 literal. It needs to be awaited.